app/assets/js/generator.js

```diff
@@ -19,36 +19,52 @@
     var tpl = doT.template($('#plugins-all-template').text());
     var pluginsAll = $('#plugins-all');
 
-    $.getJSON('https://storage.googleapis.com/generators.yeoman.io/cache.json')
-    .done(function (plugins) {
-      if (Array.isArray(plugins)) {
-        pluginsAll.html(tpl({
-          modules: plugins.sort(function (a, b) {
-            return a.stars === b.stars ? 0 : a.stars < b.stars ? 1 : -1;
-          })
-        }));
-
-        var list = new win.List('plugins-all', {
-          valueNames: [
-            'name',
-            'owner',
-            'stars',
-            'updated',
-            'downloads',
-            'description'
-          ]
-        });
-
-        if (list.listContainer) {
-          list.on('updated', function () {
-            $('.table thead').toggle(list.matchingItems.length !== 0);
-            $('#search-notfound').toggle(list.matchingItems.length === 0);
+    var lastSha = win.localStorage.getItem('sha');
+    var updatedAt = win.localStorage.getItem('updatedAt');
+
+    if (lastSha && updatedAt && Date.now() - updatedAt < 1800000) {
+      fetchAndDisplayListGenerators(lastSha);
+    } else {
+      $.getJSON('https://api.github.com/repos/yeoman/yeoman-generator-list/commits?sha=cache-generators-list')
+      .done(function (response) {
+        win.localStorage.setItem('sha', response[0].sha);
+        win.localStorage.setItem('updatedAt', Date.now());
+        fetchAndDisplayListGenerators(response[0].sha);
+      });
+    }
+
+    function fetchAndDisplayListGenerators(sha) {
+      $.getJSON(`https://cdn.rawgit.com/yeoman/yeoman-generator-list/${sha}/cache.json`)
+      .done(function (plugins) {
+        if (Array.isArray(plugins)) {
+          pluginsAll.html(tpl({
+            modules: plugins.sort(function (a, b) {
+              return a.stars === b.stars ? 0 : a.stars < b.stars ? 1 : -1;
+            })
+          }));
+
+          var list = new win.List('plugins-all', {
+            valueNames: [
+              'name',
+              'owner',
+              'stars',
+              'updated',
+              'downloads',
+              'description'
+            ]
           });
+
+          if (list.listContainer) {
+            list.on('updated', function () {
+              $('.table thead').toggle(list.matchingItems.length !== 0);
+              $('#search-notfound').toggle(list.matchingItems.length === 0);
+            });
+          }
         }
-      }
-      else {
-        pluginsAll.text('There was a problem fetching the generator list. Please try again later');
-      }
-    });
+        else {
+          pluginsAll.text('There was a problem fetching the generator list. Please try again later');
```

Comment:
Done
`      pluginsAll.html('There was a problem fetching the generator list. Please try again later.<br />Or you can use <a href="https://npms.io/search?q=keywords%3Ayeoman-generator">npms.io</a>.');
`